backend/src/apiserver/resource/resource_manager.go

```diff
@@ -558,6 +563,11 @@ func (r *ResourceManager) CreateRun(ctx context.Context, run *model.Run) (*model
 	}
 	executionSpec.SetExecutionNamespace(k8sNamespace)
 
+	// Auto-create semaphore ConfigMaps if needed
+	if err := r.ensureSemaphoreConfigMap(ctx, k8sNamespace, manifest); err != nil {
```

Comment:
Could we abstract this as a new method to the `Template` interface? Perhaps a method called `ConfigurePlatformResources` with the input being the platform spec, a `model.run`, and namespace?

My thought behind this is that `CreateRun` is pipeline engine agnostic but `ensureSemaphoreConfigMap` is very specific to Argo Workflows. So we should have it be part of the interface that pipeline engines must implement.